root/etc/rc.d/rc.local

```diff
@@ -5,7 +5,14 @@ exportKCL
 
 # setup root password
 if [[ ! -z "$KCL_FDI_ROOTPW" ]]; then
-  echo "root:$KCL_FDI_ROOTPW" | chpasswd >/dev/null
+  # remove possible double and/or single quotes surrounding password string
+  KCL_FDI_ROOTPW=`echo $KCL_FDI_ROOTPW | sed "s/\('\|\"\)//g"`
+  # if password starts with dollar sign, set an encrypted flag
+  echo $KCL_FDI_ROOTPW | grep '^\$.*$' > /dev/null
```

Comment:
Sure, I can change that. The way it was done before would work not only in specific versions of shell, but that's fine.